src/init.ts

```diff
@@ -6,8 +6,8 @@ import { isNodeEnvironment } from './utils';
 export const LooselyLazy = {
   init: (mode: 'HYDRATE' | 'RENDER') => {
     SETTINGS.CURRENT_MODE = mode;
-    setCurrent(0);
     LISTENERS.length = 0;
+    setCurrent(0);
```

Comment:
Moved resetting listeners before changing current phase as was causing an error in tests. Next PR gonna convert `LISTENERS` to a Set so we can handle them better